emc-advanced-dev/unik
# Basic Openstack Integration

# Purpose or Intent

This PR introduces OpenStack provider. It consists of three commits:
- vendor code only
- stubs for OpenStack only
- implementation of basic functions

Please consult commit messages for more details.
# Testing
- start DevStack and obtain it's IP address
- add openstack provider to the daemon-config.yml, as described here
- run daemon
- cd into this folder: 
  [Example-JavaBurekMinimal.zip](https://github.com/emc-advanced-dev/unik/files/444862/Example-JavaBurekMinimal.zip)
  and run to STAGE IMAGE + RUN INSTANCE :

```
./unik_openstack_up.sh
```

and then to DELETE IMAGE and DELETE INSTANCE:

```
./unik_openstack_down.sh
```

After the first command an image and instance should appear on your OpenStack - instance should print "BUREK" to the console 10 times.
